packages/react-magma-dom/src/components/Tag/Tag.tsx

```diff
@@ -161,70 +151,71 @@ function buildButtonTextColor(props) {
     if (props.disabled) {
       // Disabled inverse state text colors
       switch (props.color) {
-        case 'primary':
-          return `${props.theme.colors.neutral03}99`;
-        case 'highContrast':
-          return `${props.theme.colors.neutral03}99`;
+        case 'primary' || 'highContrast':
+          return transparentize(0.6, props.theme.colors.neutral100);
+
         case 'lowContrast':
-          return `${props.theme.colors.neutral08}40`;
+          return transparentize(0.7, props.theme.colors.neutral100);
+
         default:
-          return `${props.theme.colors.neutral03}99`;
+          return transparentize(0.6, props.theme.colors.neutral100);
       }
     }
     // Inverse text colors
     switch (props.color) {
-      case 'danger':
-        return `${props.theme.colors.neutral}`;
-      case 'success':
-        return `${props.theme.colors.neutral}`;
       case 'primary':
-        return `${props.theme.colors.neutral}`;
+        return props.theme.colors.primary600;
       case 'lowContrast':
-        return `${props.theme.colors.neutral08}`;
+        return props.theme.colors.tertiary500;
       case 'highContrast':
-        return `${props.theme.colors.neutral}`;
+        return props.theme.colors.neutral700;
       default:
-        return `${props.theme.colors.neutral08}`;
+        return props.theme.colors.neutral100;
     }
   } else if (props.disabled && !props.isInverse) {
     // Disabled state text colors
     switch (props.color) {
-      case 'primary':
-        return `${props.theme.colors.neutral03}99`;
-      case 'highContrast':
-        return `${props.theme.colors.neutral03}99`;
-      case 'lowContrast':
-        return `${props.theme.colors.neutral03}99`;
+      case 'primary' || 'highContrast' || 'lowContrast':
+        return transparentize(0.4, props.theme.colors.neutral500);
       default:
-        return `${props.theme.colors.neutral03}99`;
+        return transparentize(0.4, props.theme.colors.neutral500);
```

Comment:
Looks like we don't need this switch anymore, and can just `return transparentize(0.4, props.theme.colors.neutral500);`